docs/proposals/resize-when-creating-by-group.md

```diff
@@ -0,0 +1,127 @@
+
+Expand the PVC's initial capacity based on the largest capacity in specified PVCs.
+
+### Motivation
+
+One use case for pvc-autoresizer is automatically extending a PV/PVC for multiple Pod/PV pairings, such as a MySQL cluster.
+In such a cluster, if a Pod/PV fails and a PV/PVC is rebuilt for the restore process, a PV is created based on the PVC template. Even if a PV of the same size as the others is actually needed, a PV of the template's capacity is created, resulting in a case where the restore process stops due to insufficient capacity.
+
+To solve this problem, we are planning to add a new function to pvc-autoresizer to create a PV/PVC with sufficient capacity at the time of creation, when a PVC of the same group already exists.
+
+### Goals
+
+- Provide the ability to expand the PVC's initial capacity based on the largest capacity in specified PVCs.
+- The PVC for determining capacity should be user-selectable.
+- [moco](https://github.com/cybozu-go/moco) checks the PV capacity at the beginning of the restore process. It should be able to handle such cases.
+
+## Proposal
+
+Rules for grouping & initial sizing
+
+- If creating a PVC has a `resize.topolvm.io/pre-resize-group-by` annotation and the label specified in the annotation exists, existing PVCs with matching values are determined to be in the same group.
+- If no PVCs are determined to be in the same group, or if the capacity of creating PVC is larger than existing any PVC in the same group, the capacity of the original PVC is used as is.
+
+Examples are given below.
+
+```yaml
+### existing PVCs (excerpted)
+kind: PersistentVolumeClaim
+metadata:
+  name: pvc-x-1
+  labels:
+    label-hoge: group-x
+  annotations:
+    resize.topolvm.io/pre-resize-group-by: label-hoge
+spec:
+  resources:
+    requests:
+      storage: 20Gi
+
+kind: PersistentVolumeClaim
+metadata:
+  name: pvc-x-2
+  labels:
+    label-hoge: group-x
+  annotations:
+    resize.topolvm.io/pre-resize-group-by: label-hoge
+spec:
+  resources:
+    requests:
+      storage: 16Gi
+
+kind: PersistentVolumeClaim
+metadata:
+  name: pvc-y-1
+  labels:
+    label-hoge: group-y
+  annotations:
+    resize.topolvm.io/pre-resize-group-by: label-hoge
+spec:
+  resources:
+    requests:
+      storage: 30Gi
+```
+
+When creating the following new PVC, `pvc-x-1` and `pvc-x-2` with `label-hoge: group-x` are determined to be in the same group, and `pvc-y-1` is determined to be in a different group. Therefore, the PVC is created with **20Gi** based on `pvc-x-1`, which has the largest capacity.
+
+```yaml
+kind: PersistentVolumeClaim
+metadata:
+  name: pvc-x-3
+  labels:
+    label-hoge: group-x
+  annotations:
+    resize.topolvm.io/pre-resize-group-by: label-hoge
+spec:
+  resources:
+    requests:
+      storage: 10Gi
+```
+
+When creating the following new PVC, `pvc-y-1` with `label-hoge: group-y` is a candidate, but since the new PVC's size(50Gi) is larger than the existing(30Gi), the PVC will be created with **50Gi**.
```

Comment:
```suggestion
When creating the following new PVC, `pvc-y-1` with `label-hoge: group-y` is in the same group. However, since the new PVC's size(50Gi) is larger than the existing one(30Gi), the PVC is created with **50Gi**.
```